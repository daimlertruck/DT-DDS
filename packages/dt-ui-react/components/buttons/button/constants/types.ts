import { ComponentSize } from '../../../../types/componentSize';

export type ButtonVariant = 'solid' | 'outlined' | 'text';

export type ButtonColor = 'primary' | 'error' | 'neutral';

export type ButtonSize = Extract<ComponentSize, 'medium' | 'large' | 'small'>;
