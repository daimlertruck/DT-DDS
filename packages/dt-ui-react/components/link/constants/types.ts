export type LinkType = 'button' | 'text';
