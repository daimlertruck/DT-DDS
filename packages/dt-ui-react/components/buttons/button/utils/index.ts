export * from './buttonVariantColors';
export * from './buttonVariantStyles';
