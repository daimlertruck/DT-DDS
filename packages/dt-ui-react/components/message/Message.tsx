import { useState, SVGProps, ReactElement } from 'react';

import {
  CheckCircleIcon,
  CloseIcon,
  ErrorIcon,
  InfoIcon,
  WarningOutlinedIcon,
} from '../../core';
import { BaseProps } from '../../types';
import { Typography } from '../typography';

import {
  MessageStyled,
  MessageIconStyled,
  MessageButtonCloseStyled,
  MessageContentStyled,
  MessageActionStyled,
} from './Message.styled';
import { MessageType } from './types';

export interface MessageProps extends BaseProps {
  type: MessageType;
  isDismissable?: boolean;
}

type MessageIcon = (props: SVGProps<SVGSVGElement>) => ReactElement;

export const MessageIcons: Record<MessageType, MessageIcon | null> = {
  [MessageType.Error]: ErrorIcon,
  [MessageType.Info]: InfoIcon,
  [MessageType.Success]: CheckCircleIcon,
  [MessageType.Warning]: WarningOutlinedIcon,
  [MessageType.Default]: null,
};

export const Message = ({
  children,
  dataTestId,
  type,
  isDismissable = true,
}: MessageProps) => {
  const [isDismissed, setIsDismissed] = useState(false);
  const Icon = MessageIcons[type];

  const handleDismiss = () => {
    setIsDismissed(true);
  };

  if (isDismissed) return null;

  return (
    <MessageStyled data-testid={dataTestId ?? 'message'} type={type}>
      {!!Icon ? (
        <MessageIconStyled type={type}>
          <Icon data-testid='message-icon' height='16px' width='16px' />
        </MessageIconStyled>
      ) : null}
      <MessageContentStyled>{children}</MessageContentStyled>
      {isDismissable ? (
        <MessageButtonCloseStyled onClick={handleDismiss}>
          <CloseIcon height='16px' width='16px' />
        </MessageButtonCloseStyled>
      ) : null}
    </MessageStyled>
  );
};

Message.Title = ({ children }: BaseProps) => {
  return (
    <Typography color='content.heading' element='h2' fontStyles='body2Bold'>
      {children}
    </Typography>
  );
};

Message.Description = ({ children }: BaseProps) => {
  return (
    <Typography color='content.body' fontStyles='body3'>
      {children}
    </Typography>
  );
};

Message.Action = ({ children, dataTestId }: BaseProps) => (
  <MessageActionStyled data-testid={dataTestId ?? 'message-action'}>
    {children}
  </MessageActionStyled>
);
