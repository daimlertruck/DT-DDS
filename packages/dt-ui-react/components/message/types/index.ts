export enum MessageType {
  Default = 'default',
  Error = 'error',
  Info = 'info',
  Success = 'success',
  Warning = 'warning',
}
