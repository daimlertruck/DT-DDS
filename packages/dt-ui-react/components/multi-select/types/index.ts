import { ReducerActionProps } from '../reducer';

export type MultiSelectOptionValue = {
  text?: string;
  value: string;
  isRemovable?: boolean;
};

export interface MultiSelectContextState {
  state: MultiSelectOptionValue[];
  isOpen: boolean;
  name?: string;
  dispatch: React.Dispatch<ReducerActionProps>;
  setIsOpen: React.Dispatch<React.SetStateAction<boolean>>;
}

export enum Actions {
  ADD = 'ADD',
  REMOVE = 'REMOVE',
}
