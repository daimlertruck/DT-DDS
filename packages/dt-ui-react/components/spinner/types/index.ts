export type Size = 'small' | 'large';

export type SpinnerTheme = 'light' | 'dark';
