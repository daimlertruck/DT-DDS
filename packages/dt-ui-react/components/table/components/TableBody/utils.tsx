import { Children, ReactNode, cloneElement, isValidElement } from 'react';

import { TableDataCellHeader } from '../TableDataCell/TableDataCell';

export const addHeadersToDataCells = (
  dataCells: ReactNode,
  headers: ReactNode[]
) => {
  return Children.map(dataCells, (dataCell, index) => {
    if (isValidElement(dataCell)) {
      const header = headers[index];
      return cloneElement(
        dataCell,
        {},
        <>
          <TableDataCellHeader>{header}</TableDataCellHeader>
          {dataCell.props.children}
        </>
      );
    }
    return dataCell;
  });
};

export const addHeadersToDataCellsInRows = (
  rows: ReactNode,
  headers: ReactNode[]
) => {
  return Children.map(rows, (row) => {
    if (isValidElement(row)) {
      return cloneElement(
        row,
        {},
        addHeadersToDataCells(row.props.children, headers)
      );
    }
    return row;
  });
};
