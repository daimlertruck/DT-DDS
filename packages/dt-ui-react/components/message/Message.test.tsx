import { fireEvent, render, screen } from '@testing-library/react';

import { theme } from '../../themes/default';
import { withProviders } from '../../utils';
import { Link } from '../link';

import { Message } from './Message';
import { MessageType } from './types';

describe('<Message /> component', () => {
  const ProvidedMessage = withProviders(Message);
  const title = 'Title here';
  const description = 'Some description here.';

  it.each`
    type
    ${MessageType.Info}
    ${MessageType.Success}
    ${MessageType.Warning}
    ${MessageType.Error}
    ${MessageType.Default}
  `(
    'should render a Message with title & description, when type $type',
    ({ type }: { type: MessageType }) => {
      const { container } = render(
        <ProvidedMessage type={type}>
          <Message.Title>{title}</Message.Title>
          <Message.Description>{description}</Message.Description>
        </ProvidedMessage>
      );

      const icon = screen.queryByTestId('message-icon');
      if (type === MessageType.Default) {
        expect(icon).not.toBeInTheDocument();
      } else {
        expect(icon).toBeInTheDocument();
        const messageIconStyled = icon?.parentElement;
        expect(messageIconStyled).toHaveAttribute('type', type);
      }

      expect(screen.queryByTestId('message')).toHaveStyle(
        `background-color: ${
          type === MessageType.Default
            ? theme.colors.grey_10
            : theme.palette[`${type}Light_100`]
        };`
      );

      expect(container).toMatchSnapshot();
    }
  );

  it('should render a Message without title', () => {
    render(<ProvidedMessage type={MessageType.Success} />);

    const messageTitle = screen.queryByRole('heading');
    expect(messageTitle).toBeNull();
  });

  it('should dismiss the Message', () => {
    render(<ProvidedMessage type={MessageType.Warning} />);

    const messageCloseButton = screen.getByRole('button');
    expect(messageCloseButton).toBeInTheDocument();

    fireEvent.click(messageCloseButton);

    expect(screen.queryByTestId('message')).toBeNull();
  });

  it('should not render close icon when the component is not dismissable', () => {
    render(
      <ProvidedMessage isDismissable={false} type={MessageType.Default} />
    );

    expect(screen.queryByRole('button')).not.toBeInTheDocument();
  });

  it('should render a Message with an Action', () => {
    const mockOnClick = jest.fn();
    render(
      <ProvidedMessage type={MessageType.Info}>
        <Message.Action>
          <Link href='#' onClick={mockOnClick}>
            View action
          </Link>
        </Message.Action>
      </ProvidedMessage>
    );

    const messageAction = screen.getByRole('link');
    expect(messageAction).toBeInTheDocument();

    fireEvent.click(messageAction);

    expect(mockOnClick).toHaveBeenCalled();
  });
});
