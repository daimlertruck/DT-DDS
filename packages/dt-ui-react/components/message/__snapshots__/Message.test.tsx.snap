// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message /> component should render a Message with title & description, when type default 1`] = `
.emotion-0 {
  padding: 12px 16px;
  background-color: #F8F9FA;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border-radius: 2px;
  border: 1px dashed #F8F9FA;
  width: 100%;
  height: 100%;
  gap: 8px;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  text-align: start;
  gap: 2px;
}

.emotion-4 {
  font-weight: 700;
  font-size: 0.875rem;
  line-height: 1.25rem;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  color: #323232;
}

.emotion-6 {
  font-weight: 400;
  font-size: 0.75rem;
  line-height: 1rem;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  color: #7C7C7D;
}

.emotion-8 {
  color: #7C7C7D;
  border: 0;
  cursor: pointer;
  background: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-8:hover {
  background-color: #FFFFFF;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-testid="message"
    type="default"
  >
    <div
      class="emotion-2 emotion-3"
    >
      <h2
        class="emotion-4 emotion-5"
        color="textPrimary"
        data-testid="typography"
      >
        Title here
      </h2>
      <p
        class="emotion-6 emotion-5"
        color="textSecondary"
        data-testid="typography"
      >
        Some description here.
      </p>
    </div>
    <button
      class="emotion-8 emotion-9"
    >
      <svg
        fill="none"
        height="16px"
        viewBox="0 0 24 24"
        width="16px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19 6.41L17.59 5L12 10.59L6.41 5L5 6.41L10.59 12L5 17.59L6.41 19L12 13.41L17.59 19L19 17.59L13.41 12L19 6.41Z"
          fill="currentColor"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`<Message /> component should render a Message with title & description, when type error 1`] = `
.emotion-0 {
  padding: 12px 16px;
  background-color: #FCEEED;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border-radius: 2px;
  border: 1px dashed #FF4A4A;
  width: 100%;
  height: 100%;
  gap: 8px;
}

.emotion-2 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #FF4A4A;
  margin-top: 2px;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  text-align: start;
  gap: 2px;
}

.emotion-6 {
  font-weight: 700;
  font-size: 0.875rem;
  line-height: 1.25rem;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  color: #323232;
}

.emotion-8 {
  font-weight: 400;
  font-size: 0.75rem;
  line-height: 1rem;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  color: #7C7C7D;
}

.emotion-10 {
  color: #7C7C7D;
  border: 0;
  cursor: pointer;
  background: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-10:hover {
  background-color: #FFFFFF;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-testid="message"
    type="error"
  >
    <div
      class="emotion-2 emotion-3"
      type="error"
    >
      <svg
        color="#FF4A4A"
        data-testid="message-icon"
        fill="none"
        height="16px"
        viewBox="0 0 24 24"
        width="16px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19.0711 19.0711C17.1957 20.9464 14.6522 22 12 22C9.34784 22 6.8043 20.9464 4.92893 19.0711C3.05357 17.1957 2 14.6522 2 12C2 9.34784 3.05357 6.8043 4.92893 4.92893C6.8043 3.05357 9.34784 2 12 2C14.6522 2 17.1957 3.05357 19.0711 4.92893C20.9464 6.8043 22 9.34784 22 12C22 14.6522 20.9464 17.1957 19.0711 19.0711Z"
          fill="currentColor"
        />
        <path
          d="M12.8797 13.6863L13.1157 6.48828H10.8997L11.0997 13.6863H12.8797ZM11.9907 17.5123C12.1621 17.5181 12.333 17.4894 12.4932 17.4281C12.6534 17.3667 12.7997 17.2739 12.9234 17.1551C13.0471 17.0362 13.1458 16.8938 13.2136 16.7362C13.2813 16.5786 13.3168 16.4091 13.318 16.2375C13.3191 16.066 13.2859 15.8959 13.2202 15.7375C13.1545 15.579 13.0577 15.4353 12.9356 15.3148C12.8135 15.1944 12.6684 15.0996 12.5091 15.0361C12.3497 14.9727 12.1792 14.9418 12.0077 14.9453C11.6733 14.9521 11.3547 15.0892 11.1198 15.3274C10.8849 15.5656 10.7522 15.886 10.75 16.2205C10.7478 16.555 10.8762 16.8772 11.1079 17.1184C11.3397 17.3597 11.6564 17.501 11.9907 17.5123Z"
          fill="white"
        />
      </svg>
    </div>
    <div
      class="emotion-4 emotion-5"
    >
      <h2
        class="emotion-6 emotion-7"
        color="textPrimary"
        data-testid="typography"
      >
        Title here
      </h2>
      <p
        class="emotion-8 emotion-7"
        color="textSecondary"
        data-testid="typography"
      >
        Some description here.
      </p>
    </div>
    <button
      class="emotion-10 emotion-11"
    >
      <svg
        fill="none"
        height="16px"
        viewBox="0 0 24 24"
        width="16px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19 6.41L17.59 5L12 10.59L6.41 5L5 6.41L10.59 12L5 17.59L6.41 19L12 13.41L17.59 19L19 17.59L13.41 12L19 6.41Z"
          fill="currentColor"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`<Message /> component should render a Message with title & description, when type info 1`] = `
.emotion-0 {
  padding: 12px 16px;
  background-color: #E6F5FF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border-radius: 2px;
  border: 1px dashed #008DFC;
  width: 100%;
  height: 100%;
  gap: 8px;
}

.emotion-2 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #008DFC;
  margin-top: 2px;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  text-align: start;
  gap: 2px;
}

.emotion-6 {
  font-weight: 700;
  font-size: 0.875rem;
  line-height: 1.25rem;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  color: #323232;
}

.emotion-8 {
  font-weight: 400;
  font-size: 0.75rem;
  line-height: 1rem;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  color: #7C7C7D;
}

.emotion-10 {
  color: #7C7C7D;
  border: 0;
  cursor: pointer;
  background: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-10:hover {
  background-color: #FFFFFF;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-testid="message"
    type="info"
  >
    <div
      class="emotion-2 emotion-3"
      type="info"
    >
      <svg
        data-testid="message-icon"
        fill="currentColor"
        height="16px"
        viewBox="0 0 24 24"
        width="16px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12C2 17.52 6.48 22 12 22C17.52 22 22 17.52 22 12C22 6.48 17.52 2 12 2ZM13 17H11V11H13V17ZM13 9H11V7H13V9Z"
        />
      </svg>
    </div>
    <div
      class="emotion-4 emotion-5"
    >
      <h2
        class="emotion-6 emotion-7"
        color="textPrimary"
        data-testid="typography"
      >
        Title here
      </h2>
      <p
        class="emotion-8 emotion-7"
        color="textSecondary"
        data-testid="typography"
      >
        Some description here.
      </p>
    </div>
    <button
      class="emotion-10 emotion-11"
    >
      <svg
        fill="none"
        height="16px"
        viewBox="0 0 24 24"
        width="16px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19 6.41L17.59 5L12 10.59L6.41 5L5 6.41L10.59 12L5 17.59L6.41 19L12 13.41L17.59 19L19 17.59L13.41 12L19 6.41Z"
          fill="currentColor"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`<Message /> component should render a Message with title & description, when type success 1`] = `
.emotion-0 {
  padding: 12px 16px;
  background-color: #E9F6EA;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border-radius: 2px;
  border: 1px dashed #21A330;
  width: 100%;
  height: 100%;
  gap: 8px;
}

.emotion-2 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #21A330;
  margin-top: 2px;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  text-align: start;
  gap: 2px;
}

.emotion-6 {
  font-weight: 700;
  font-size: 0.875rem;
  line-height: 1.25rem;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  color: #323232;
}

.emotion-8 {
  font-weight: 400;
  font-size: 0.75rem;
  line-height: 1rem;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  color: #7C7C7D;
}

.emotion-10 {
  color: #7C7C7D;
  border: 0;
  cursor: pointer;
  background: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-10:hover {
  background-color: #FFFFFF;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-testid="message"
    type="success"
  >
    <div
      class="emotion-2 emotion-3"
      type="success"
    >
      <svg
        data-testid="message-icon"
        fill="currentColor"
        height="16px"
        viewBox="0 0 24 24"
        width="16px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12C2 17.52 6.48 22 12 22C17.52 22 22 17.52 22 12C22 6.48 17.52 2 12 2ZM10 17L5 12L6.41 10.59L10 14.17L17.59 6.58L19 8L10 17Z"
        />
      </svg>
    </div>
    <div
      class="emotion-4 emotion-5"
    >
      <h2
        class="emotion-6 emotion-7"
        color="textPrimary"
        data-testid="typography"
      >
        Title here
      </h2>
      <p
        class="emotion-8 emotion-7"
        color="textSecondary"
        data-testid="typography"
      >
        Some description here.
      </p>
    </div>
    <button
      class="emotion-10 emotion-11"
    >
      <svg
        fill="none"
        height="16px"
        viewBox="0 0 24 24"
        width="16px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19 6.41L17.59 5L12 10.59L6.41 5L5 6.41L10.59 12L5 17.59L6.41 19L12 13.41L17.59 19L19 17.59L13.41 12L19 6.41Z"
          fill="currentColor"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`<Message /> component should render a Message with title & description, when type warning 1`] = `
.emotion-0 {
  padding: 12px 16px;
  background-color: #FFFAE6;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border-radius: 2px;
  border: 1px dashed #FBD633;
  width: 100%;
  height: 100%;
  gap: 8px;
}

.emotion-2 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #FBD633;
  margin-top: 2px;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  text-align: start;
  gap: 2px;
}

.emotion-6 {
  font-weight: 700;
  font-size: 0.875rem;
  line-height: 1.25rem;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  color: #323232;
}

.emotion-8 {
  font-weight: 400;
  font-size: 0.75rem;
  line-height: 1rem;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  color: #7C7C7D;
}

.emotion-10 {
  color: #7C7C7D;
  border: 0;
  cursor: pointer;
  background: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-10:hover {
  background-color: #FFFFFF;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-testid="message"
    type="warning"
  >
    <div
      class="emotion-2 emotion-3"
      type="warning"
    >
      <svg
        data-testid="message-icon"
        fill="currentColor"
        height="16px"
        viewBox="0 0 24 24"
        width="16px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 21.5H23L12 2.5L1 21.5ZM13 18.5H11V16.5H13V18.5ZM13 14.5H11V10.5H13V14.5Z"
        />
      </svg>
    </div>
    <div
      class="emotion-4 emotion-5"
    >
      <h2
        class="emotion-6 emotion-7"
        color="textPrimary"
        data-testid="typography"
      >
        Title here
      </h2>
      <p
        class="emotion-8 emotion-7"
        color="textSecondary"
        data-testid="typography"
      >
        Some description here.
      </p>
    </div>
    <button
      class="emotion-10 emotion-11"
    >
      <svg
        fill="none"
        height="16px"
        viewBox="0 0 24 24"
        width="16px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19 6.41L17.59 5L12 10.59L6.41 5L5 6.41L10.59 12L5 17.59L6.41 19L12 13.41L17.59 19L19 17.59L13.41 12L19 6.41Z"
          fill="currentColor"
        />
      </svg>
    </button>
  </div>
</div>
`;
